File: src/main/java/com/rcoverzola/ejers/ValidateStorePass.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.rcoverzola.ejers;

import java.util.Scanner;

/**
 *
 * @author Roger Coverzola Bogado
 */
public class ValidateStorePass {
    private String[] store_pass;
    private Scanner scanner;
    private boolean is_exit = true;
    private String[] MESSAJE = {"Correcta!!", "Incorrecta!!", "Ingrese la clave: ", "Intente nuevamente"};
       
    public ValidateStorePass(String... arr_pass){
        this.scanner = new Scanner(System.in);
        this.store_pass = new String[arr_pass.length];
        System.arraycopy(arr_pass, 0, this.store_pass, 0, arr_pass.length);
    }
    
    private boolean validatePass(String pass){
        for(String value_pass: this.store_pass){
            if (value_pass.equalsIgnoreCase(pass)) {
                   return value_pass.equalsIgnoreCase(pass);
            }
        }
        return false;
    }
    
    public void isValidate(){
        
        while(this.is_exit){
                System.out.println(this.MESSAJE[2]);
            String pass = this.scanner.next();
            if (validatePass(pass)){
                System.out.println(this.MESSAJE[0]);
                this.is_exit=false;
            }else{
                System.out.println(this.MESSAJE[1]);
                System.out.println(this.MESSAJE[3]);
            }
        }       
        this.scanner.close();
    }
    
    
}

File: src/ValidateStorePass.ts
export class ValidateStorePass {
    private store_pass: string[];
    private scanner: any;
    private is_exit: boolean = true;
    private static readonly MESSAJE: string[] = [
        "Correcta!!", 
        "Incorrecta!!", 
        "Ingrese la clave: ", 
        "Intente nuevamente"
    ];
       
    constructor(...arr_pass: string[]) {
        this.scanner = require('readline').createInterface({
            input: process.stdin,
            output: process.stdout
        });
        this.store_pass = [...arr_pass];
    }
    
    private validatePass(pass: string): boolean {
        for (const value_pass of this.store_pass) {
            if (value_pass.toLowerCase() === pass.toLowerCase()) {
                return true;
            }
        }
        return false;
    }
    
    public isValidate(): void {
        const askPassword = () => {
            if (!this.is_exit) {
                this.scanner.close();
                return;
            }
            
            console.log(ValidateStorePass.MESSAJE[2]);
            
            this.scanner.question('', (pass: string) => {
                if (this.validatePass(pass)) {
                    console.log(ValidateStorePass.MESSAJE[0]);
                    this.is_exit = false;
                } else {
                    console.log(ValidateStorePass.MESSAJE[1]);
                    console.log(ValidateStorePass.MESSAJE[3]);
                }
                
                askPassword();
            });
        };
        
        askPassword();
    }
}
